test/UI/page_object/forms_page.py

```diff
@@ -1,61 +1,135 @@
+from dataclasses import asdict
+from typing import List
+
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 from selenium.webdriver.support import expected_conditions as EC
 
-from test.UI.locators import forms_page_locators
+from core.decorators import handle_none_argument
+from core.enums.hobby import Hobby
+from core.enums.user import User, convert_user_object_to_dict
 from test.UI.page_object.base_page import BasePage
+from utils.base_utils import DEFAULT_EXPLICIT_WAIT
 
 
 class FormsPage(BasePage):
+    PRACTICE_FORM = (By.XPATH, "//span[text()='Practice Form']")
+    FIRST_NAME_FIELD = (By.CSS_SELECTOR, "#firstName")
+    LAST_NAME_FIELD = (By.CSS_SELECTOR, "#lastName")
+    PHONE_FIELD = (By.CSS_SELECTOR, "#userNumber")
+    SUBMIT_BUTTON = (By.CSS_SELECTOR, "button#submit")
+    GENDER_RADIO_BUTTON = (By.XPATH, "//div[@id='genterWrapper']//input[@value='{}']/..")
+    DATE_OF_BIRTH_FIELD = (By.CSS_SELECTOR, "#dateOfBirthInput")
+    SUBJECTS_FIELD = (By.CSS_SELECTOR, "#subjectsContainer")
+    EMAIL_FIELD = (By.CSS_SELECTOR, "#userEmail")
+    HOBBY_CHECKBOX = (By.XPATH, "//label[contains(@for,'hobbies-checkbox-') and contains(text(), '{}')]")
+    CURRENT_ADDRESS_FIELD = (By.XPATH, "//textarea[@placeholder='Current Address']")
+
+    WINDOW_OF_SUBMITTED_DATA = (By.XPATH, "//div[text()='Thanks for submitting the form']")
 
     # Student Registration Form
-    def click_practice_form(self):
-        self.click(forms_page_locators.PRACTICE_FORM)
-
-    def fill_first_name(self, first_name):
-        self.send_keys(forms_page_locators.FIRST_NAME_FIELD, first_name)
-
-    def fill_last_name(self, last_name):
-        self.send_keys(forms_page_locators.LAST_NAME_FIELD, last_name)
-
-    def enter_mobile_phone(self, phone):
-        self.send_keys(forms_page_locators.PHONE_FIELD, phone)
-
-    def click_gender_radio_button(self, gender):
-        self.click((By.XPATH, "//div[@id='genterWrapper']//input[@value='{}']/..".format(gender)))
-
-    def click_submit_button(self):
+    def registrate_user(self, user: User):
+        self.enter_first_name(user.first_name)
+        self.enter_last_name(user.last_name)
+        self.enter_email(user.email)
+        self.click_gender_radio_button(user.gender)
+        self.enter_mobile_phone(user.phone)
+        self.enter_date_of_birth(user.date_of_birth)
+        self.enter_subjects(user.subjects)
+        self.click_hobby(user.hobbies)
+        self.enter_current_address(user.current_address)
+        self.enter_state(user.state)
+        self.enter_city(user.state)
```

Comment:
Applying 'YAGNI' principle: you're not going to use all these `enter_whatever` methods, so don't write them. Make the `register` method to `send_keys` directly to the controls. When you will need to enter only subject, only then you should extract ` self.send_keys(self.SUBJECTS_FIELD, user.subjects)` to the `enter_subjects` method.

```suggestion
        self.send_keys(self.DATE_OF_BIRTH_FIELD, user.date_of_birth)
        self.send_keys(self.SUBJECTS_FIELD, user.subjects)
        self.click_hobby(user.hobbies) # I think this is good for readability
        self.send_keys(self.CURRENT_ADDRESS_FIELD, user.current_address)
```

Reply:
done